### ISSUE TITLE
Duplicate alerts in list

### ISSUE THREAD

**Issue by ipstatic** [prometheus/alertmanager#805]
When testing 0.6.2 we noticed duplicate alerts showing in the listing when more than one receiver is configured. Using Git bisect, it appears to be introduced in this [commit](https://github.com/prometheus/alertmanager/commit/1e34f29532a251b728ddf304ebccd9234fcbd81a). If you run the same configuration before this commit, you do not see the duplication issue.

![screen shot 2017-05-18 at 10 17 11 am](https://cloud.githubusercontent.com/assets/332617/26208478/2f337974-3bb8-11e7-88a2-09e9d555bae6.png)

![screen shot 2017-05-18 at 10 17 19 am](https://cloud.githubusercontent.com/assets/332617/26207663/2a23f58c-3bb6-11e7-90ea-8e5ea3648141.png)

Here is the JSON response from `api/v1/alerts/groups` for pre commit:

[pre.txt](https://github.com/prometheus/alertmanager/files/1011239/pre.txt)

and post:

[post.txt](https://github.com/prometheus/alertmanager/files/1011240/post.txt)

Finally here is the configuration I used to replicate the issue:

[prometheus.txt](https://github.com/prometheus/alertmanager/files/1011251/prometheus.txt)
[alertmanager.txt](https://github.com/prometheus/alertmanager/files/1011252/alertmanager.txt)

**Comment by jjneely** [comment]
So, there is a known issue #539 that causes duplicate alerts to appear on the UI when routing alerts to two different receivers.  This should be fixed with the UI design.  But testing this version....yes now I have 4 alerts in the UI when routing alerts to 2 receivers.

This is not something I can expose my operations teams to.

**Comment by mxinden** [comment]
@ipstatic as @jjneely pointed out, this will be fixed with the UI rewrite. The only thing holding us back from releasing the new UI right now is #806, which should be merged in the upcoming days. 

Thanks for reporting the issue. I will close here. Feel free to reopen if you have any further questions.

**Comment by ipstatic** [comment]
@mxinden Actually @jjneely was pointing out this is not the same issue. The API is duplicating data per "group" after this filtering was added. Please look at my pre and post dumps of this data.

**Comment by jjneely** [comment]
Yes, this is now significantly worse.  We've introduced more bugs, not kept the same.

**Comment by mxinden** [comment]
@ipstatic I am sorry for closing so quickly here.  I will look into it tomorrow.

**Comment by r4j4h** [comment]
I don't want to pollute with more noise but I seem to be running into this as well and wanted to contribute something interesting that may help those debugging.

My afflicted alert uses variables and they only show up in the first one, but not the duplicated alerts.

IOTW, something like

`hi "{{ $labels.name }}"`

Shows up in first alert as 

`hi "foo"`

while the duplicated alerts get

`hi ""`

Again sorry if this is a separate issue, it feels similar and I am hoping this aids those working on a solution.

**Comment by mxinden** [comment]
@ipstatic @jjneely & @r4j4h

We have just released a new version of Alertmanager (See [v0.7.0-rc.0](https://github.com/prometheus/alertmanager/releases/tag/v0.7.0-rc.0). Can you test if the described issue still exists in the new release candidate? 

If so we will fix it in v0.7. Right now there would be no plans to backport the fix to v0.6 as the two are too different.

**Comment by ipstatic** [comment]
@mxinden It does not appear in 0.7. Any ideas when this version will be released?

**Comment by jjneely** [comment]
Yeah, it does appear fixed in 0.7 so that will be our next target to deploy rather than 0.6.x.

I guess the real question I'd like to pose is how can we test multiple receivers going forward?  Is that part of the automated testing / can it be?

**Comment by mxinden** [comment]
@ipstatic I will update the [v0.7](https://github.com/prometheus/alertmanager/milestone/4) milestone today with all the issues that still need to be fixed before we can release. That's the best I can do.

@jjneely I would have to redirect this to @fabxc. The initial duplication issue is fixed due to the fact, that the new UI is using the `/alerts` endpoint, which is not aware of AlertGroups or receivers.

**Comment by fabxc** [comment]
Did the duplication issue come from `/alerts/groups`? As [this commit](https://github.com/prometheus/alertmanager/commit/1e34f29532a251b728ddf304ebccd9234fcbd81a) was reported to have introduced it, I assume yes.

So a) we have to fix it or b) we should remove `/alerts/groups`. Generally fine with b) but I think we have currently no other way to inspect the internal state of our routing tree for potential debugging, which could be a problem.

@stuartnelson3 

@jjneely what do you mean by "testing multiple receivers" specifically?

**Comment by r4j4h** [comment]
@mxinden It looks like my issue on the repeated alerts not being hydrated is still a problem when running v0.7.0-rc.0.

On the positive side there is no negative change, it behaves the same.

Since it appears the main issue is fixed, should I create a new Issue regarding my variables-used-once problem?

**Comment by r4j4h** [comment]
Actually more to report so editing my issue (just letting you know incase you try to respond wihle I do so, will delete this entry after)

**Comment by mxinden** [comment]
@r4j4h Thanks for the positive feedback. Could you create a new issue for your feature request?

**Comment by w0rm** [comment]
@mxinden @fabxc I believe this is fixed in the new UI because we're not using /alerts/groups anymore?

**Comment by fabxc** [comment]
Yes, should be fixed.
